### DISCUSSION TITLE
SBAS-InSAR Pre-processing

### DISCUSSION THREAD

**Discussion by Abdallahbalushi** [insarlab/MintPy#1115] [Q&A]
I am trying to do a preprocessing for sentinel-1 images acquired from 2017 to 2022 to monitor surface deformation over an oilfield. I am in a necessary need for an assistance in undertaking this step. I want to do it with SNAP, however, it is time consuming.

**Comment by Asmit-iirs** [comment]
You can use the ASF Hyp3 facility to process the interferograms and use MintPy thereafter to do the TS analysis

**Reply by forrestfwilliams** [reply]
@Abdallahbalushi HyP3 is a great option if you want to get up and running with MintPy quickly. [This page](https://storymaps.arcgis.com/stories/68a8a3253900411185ae9eb6bb5283d3) walks you through how to order HyP3 InSAR products.

**Reply by ditafaith** [reply]
Hi @forrestfwilliams @Asmit-iirs 

Suppose, I have generated smallbaselineApp.cfg after ran `smallbaselineApp.py -g`, but there is no guidance on how to produce *txt file as shown in the example file for those who want to process from HyP3. Could you help me how to produce text file.

For example, in this [link](https://mintpy.readthedocs.io/en/latest/demo_dataset/) only showed to execute 
`smallbaselineApp.py RidgecrestSenDT71.txt` but it is difficult to understand where the file RidgecrestSenDT71.txt comes from and how to generate this text file.

**Reply by Asmit-iirs** [reply]
Hey @ditafaith I am sorry I did not see this before but you can use the following to write the config file

```
# utils function
def configure_template_file(outName, CONFIG_TXT): 
    """Write configuration files for MintPy to process HyP3 product"""
    if os.path.isfile(outName):
        with open(outName, "w") as fid:
            fid.write(CONFIG_TXT)
        print('write configuration to file: {}'.format(outName))

    else:
        with open(outName, "a") as fid:
            fid.write("\n" + CONFIG_TXT)
        print('add the following to file: \n{}'.format(outName))
```

and 

```
CONFIG_TXT = f'''# vim: set filetype=cfg:
mintpy.load.processor         = hyp3
mintpy.compute.maxMemory = auto #[float > 0.0], auto for 4, max memory to allocate in GB
ORBIT_DIRECTION = descending
##---------interferogram datasets:
mintpy.load.unwFile          = {hyp3_dir}/*/*unw_phase_clipped.tif
mintpy.load.corFile          = {hyp3_dir}/*/*corr_clipped.tif
##---------geometry datasets:
mintpy.load.demFile          = {hyp3_dir}/*/*dem_clipped.tif
mintpy.load.incAngleFile     = {hyp3_dir}/*/*lv_theta_clipped.tif
mintpy.load.waterMaskFile    = {hyp3_dir}/*/*water_mask_clipped.tif
mintpy.reference.lalo        = auto #[31.8,130.8 / auto] #[N,E]
mintpy.reference.date        = 20170321  #[reference_date.txt / 20090214 / no], auto for reference_date.txt
mintpy.solidEarthTides       = yes    #[yes / no], auto for no
mintpy.deramp                = quadratic    #[no / linear / quadratic], auto for no - no ramp will be removed
mintpy.save.hdfEos5          = yes     #[yes / no], auto for no, save time-series to HDF-EOS5 format
mintpy.plot.dpi              = 400  #[int], auto for 150, number of dots per inch (DPI)
mintpy.geocode.laloStep      = auto  #[-0.000555556,0.000555556 / None], auto for None, output resolution in degree
mintpy.unwrapError.method    = auto  #[bridging / phase_closure / bridging+phase_closure / no], auto for no
mintpy.networkInversion.minTempCoh  = auto #[0.0-1.0], auto for 0.7, min temporal coherence for mask
'''
print(CONFIG_TXT)
configName = os.path.join(work_dir, "{}.txt".format(proj_name))
configure_template_file(configName, CONFIG_TXT)
```

execute the first part while importing the packages and then the second. I think it will do

**Reply by ditafaith** [reply]
Hi @Asmit-iirs thanks for your response. Yes, I got the results after ran the second one.
Now, I would like to know how to use the result in GBIS. Since they asked to convert output from mintpy should in double type format, instead single format.

**Reply by Asmit-iirs** [reply]
I don't have any experience working with GBIS at the moment. It could be helpful if you tell me what kind of inputs are required in GBIS.

**Reply by ditafaith** [reply]
@Asmit-iirs 
I got the output from mintpy and convert them using `save_gbis.py`. 
However, [the results](https://drive.google.com/file/d/1aYPFPuHk9dM2LdN74Qr2yn9ovm-tlDIh/view?usp=sharing) look like not compatible with the GBIS in matlab. 
Since, Iam not familiar with Matlab, then I am looking for the way to change it.

![Screenshot1](https://github.com/insarlab/MintPy/assets/126780178/ed71a4aa-f534-4a32-8ae1-a877cfcf5a81)

**Reply by Asmit-iirs** [reply]
@ditafaith 
Have you tried save_gbis.py -h and changing the parameters?

**Reply by ditafaith** [reply]
It has been resolved.. thanks for the attention.
